tests/jsunit/json_test.js

```diff
@@ -257,3 +257,57 @@ function test_json_dropdown_image() {
     delete Blockly.Msg['ALTTEXT'];
   }
 }
+
+function test_defineBlocksWithJsonArray_nullItem() {
```

Comment:
if there isn't a test where the typename is empty  or null (didn't see one in the immediate vicinity of this test... but also didn't look very hard), would you add one?

Reply:
Added new test towards the top.